castor-service/src/test/java/io/carbynestack/castor/service/persistence/cache/ReservationCachingServiceTest.java

```diff
@@ -58,282 +73,434 @@ public void setUp() {
+        .splitBefore(existingFragment, requestedStartIndex);
+    verify(tupleChunkFragmentStorageServiceMock, times(1))
+        .splitAt(existingFragment, requestedStartIndex + requestedLength);
+    verify(consumptionCachingServiceMock)
+        .keepConsumption(anyLong(), eq(tupleType), eq(requestedLength));
+  }
 
-    when(reservationElementMock.getTupleChunkId()).thenReturn(chunkId);
-    when(reservationElementMock.getStartIndex()).thenReturn(tupleStartIndex);
-    when(reservationElementMock.getReservedTuples()).thenReturn(requestedTuples);
-    when(reservationMock.getTupleType()).thenReturn(tupleType);
-    when(reservationMock.getReservations()).thenReturn(singletonList(reservationElementMock));
-    when(tupleChunkMetaDataStorageServiceMock.getTupleChunkData(chunkId)).thenReturn(metaDataMock);
-    when(metaDataMock.getStatus()).thenReturn(ActivationStatus.UNLOCKED);
-    when(metaDataMock.getTupleType()).thenReturn(tupleType);
-    when(metaDataMock.getReservedMarker()).thenReturn(tupleStartIndex);
-    when(metaDataMock.getNumberOfTuples()).thenReturn(totalTuplesInChunk);
+  @Test
+  public void
+      givenNoReservationReceivedInTime_whenGetReservationWithRetry_thenThrowCastorServiceException()
+          throws ExecutionException, InterruptedException, TimeoutException {
+    String reservationId = "testReservationId";
+    TupleType tupleType = MULTIPLICATION_TRIPLE_GFP;
+    long tupleCount = 42;
+    long retryDelay = 5;
+    long reservationTimeout = 0;
+    when(castorSlaveServicePropertiesMock.getRetryDelay()).thenReturn(retryDelay);
+    when(castorSlaveServicePropertiesMock.getWaitForReservationTimeout())
+        .thenReturn(reservationTimeout);
+    Future<Reservation> reservationFutureMock = mock(Future.class);
+
+    when(executorServiceMock.submit(any(WaitForReservationCallable.class)))
+        .thenReturn(reservationFutureMock);
+    when(reservationFutureMock.get(reservationTimeout, TimeUnit.MILLISECONDS))
+        .thenThrow(new InterruptedException("expected"));
 
     CastorServiceException actualCse =
         assertThrows(
             CastorServiceException.class,
-            () -> reservationCachingService.keepReservation(reservationMock));
-
+            () ->
+                reservationCachingService.getReservationWithRetry(
+                    reservationId, tupleType, tupleCount));
     assertEquals(
-        String.format(
-            NOT_ENOUGH_TUPLES_IN_CHUNK_EXCEPTION_MSG,
-            requestedTuples,
-            totalTuplesInChunk - tupleStartIndex),
+        String.format(NO_RELEASED_TUPLE_RESERVATION_EXCEPTION_MSG, reservationId),
         actualCse.getMessage());
   }
 
+  @Test
+  public void givenSuccessfulRequest_whenGetReservationWithRetry_thenReturnExpectedReservation()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    String reservationId = "testReservationId";
+    TupleType tupleType = MULTIPLICATION_TRIPLE_GFP;
```

Comment:
Used on only once in remainder of body -> redundant

Reply:
see discussion [above](https://github.com/carbynestack/castor/pull/29#discussion_r868054674)